client/server/boot/index.js

```diff
@@ -31,6 +32,15 @@ export default function setup() {
 	if ( 'development' === process.env.NODE_ENV ) {
 		require( 'calypso/server/bundler' )( app );
 
+		// When mocking WordPress.com to point locally, wordpress.com/wp-login.php will hit Calypso.
+		// redirect traffic to de.wordpress.com to hit the real backend.
+		if ( process.env.MOCK_WORDPRESSDOTCOM === '1' ) {
+			app.use(
+				'/wp-login.php',
+				createProxyMiddleware( { target: 'https://de.wordpress.com/wp-login.php' } )
+			);
+		}
```

Comment:
How does this work? Won't `https://de.wordpress.com/wp-login.php` redirect to `https://wordpress.com/log-in/de` itself?

I think logging in just won't work in this context. Perhaps we should just add a mechanism to bypass some routes and mark them as "non-working in this mocked/sandboxed setup"?

Reply:
I think it only redirects GET requests! It works for POST requests. You can login using this method just fine.

Reply:
Got it - that's a clever hack! It's worth adding a comment to explain what we're doing.

Reply:
There is a comment. Would you like me to improve on it?

Reply:
I think that's clear enough actually - thanks!